doc/GETTING_STARTED.md

```diff
@@ -0,0 +1,482 @@
+# Getting Started
+
+Welcome to libp2p, this guide will walk you through setting up a fully functional libp2p node 🚀
+
+- [Getting Started](#getting-started)
+  - [Install](#install)
+  - [Configuring libp2p](#configuring-libp2p)
+    - [Basic setup](#basic-setup)
+      - [Transports](#transports)
+      - [Connection Encryption](#connection-encryption)
+      - [Multiplexing](#multiplexing)
+      - [Running Libp2p](#running-libp2p)
+    - [Custom setup](#custom-setup)
+      - [Peer Discovery](#peer-discovery)
+      - [Peer Routing](#peer-routing)
+      - [Content Routing](#content-routing)
+    - [Pubsub](#pubsub)
+  - [What is next](#what-is-next)
+
+## Install
+
+The first step is to install libp2p in your project:
+
+```sh
+npm install libp2p
+```
+
+## Configuring libp2p
+
+If you're new to libp2p, we recommend configuring your node in stages, as this can make troubleshooting configuration issues much easier. In this guide, we'll do just that. If you're more experienced with libp2p, you may wish to jump to the [Configuration readme](./CONFIGURATION.md).
+
+### Basic setup
+
+Now that we have libp2p installed, let's configure the absolute minimum needed to get your node running. The only modules libp2p requires are a [**Transport**][transport] and [**Crypto**][crypto] module. Moreover, we recommend that a basic setup should also have a [**Stream Multiplexer**](streamMuxer) configured. Let's start by setting up a Transport.
+
+#### Transports
+
+Libp2p uses Transports to establish connections between peers over the network. You can configure 1 Transport, or as many as you like. Supporting more Transports will improve the ability for other nodes on the network to communicate with you.
+
+You should select Transports according to the runtime where your application will run. You can see a list of some of the available Transports in the [configuration readme](./CONFIGURATION.md#transport). We are going to install `libp2p-websockets`, as it can be used in both Node.js and the browser.
```

Comment:
```suggestion
You should select Transports according to the runtime of your application (Node.js / browser). You can see a list of some of the available Transports in the [configuration readme](./CONFIGURATION.md#transport). We are going to install `libp2p-websockets`, as it can be used in both Node.js and the browser.
```